pyanaconda/modules/storage/partitioning/automatic_partitioning.py

```diff
@@ -70,40 +66,85 @@ def _configure_partitioning(self, storage):
         """
         log.debug("Executing the automatic partitioning.")
 
-        # Create the auto partitioning proxy.
-        auto_part_proxy = STORAGE.get_proxy(AUTO_PARTITIONING)
+        # Get the partitioning scheme.
+        scheme = self._request.partitioning_scheme
 
         # Set the filesystem type.
-        fstype = auto_part_proxy.FilesystemType
+        fstype = self._request.file_system_type
 
         if fstype:
             storage.set_default_fstype(fstype)
 
+        # Get the encryption configuration.
+        encrypted = self._request.encrypted
+
+        # Get LUKS format args.
+        luks_format_args = self._get_luks_format_args(self._storage, self._request)
+
         # Set the default pbkdf args.
-        pbkdf_args = self._luks_format_args.get('pbkdf_args', None)
+        pbkdf_args = luks_format_args.get('pbkdf_args', None)
 
         if pbkdf_args and not luks_data.pbkdf_args:
             luks_data.pbkdf_args = pbkdf_args
 
         # Set the minimal entropy.
-        min_luks_entropy = self._luks_format_args.get('min_luks_entropy', None)
+        min_luks_entropy = luks_format_args.get('min_luks_entropy', None)
 
         if min_luks_entropy is not None:
             luks_data.min_entropy = min_luks_entropy
 
         # Get the autopart requests.
-        requests = self._get_autopart_requests(storage)
+        requests = self._get_autopart_requests(storage, self._request.excluded_mount_points)
 
         # Do the autopart.
-        self._do_autopart(storage, self._scheme, requests, self._encrypted, self._luks_format_args)
+        self._do_autopart(storage, scheme, requests, encrypted, luks_format_args)
 
-    def _get_autopart_requests(self, storage):
+    @staticmethod
+    def _get_luks_format_args(storage, request):
+        """Arguments for the LUKS format constructor.
+
+        :param storage: blivet.Blivet instance
+        :param request: a partitioning request
+        :return: a dictionary of arguments
+        """
+        if not request.encrypted:
+            return {}
+
+        luks_version = request.luks_version or storage.default_luks_version
+        escrow_cert = storage.get_escrow_certificate(request.escrow_certificate)
+
+        pbkdf_args = get_pbkdf_args(
+            luks_version=luks_version,
+            pbkdf_type=request.pbkdf or None,
+            max_memory_kb=request.pbkdf_memory,
+            iterations=request.pbkdf_iterations,
+            time_ms=request.pbkdf_time
+        )
+
+        return {
+            "passphrase": request.passphrase,
+            "cipher": request.cipher,
+            "luks_version": luks_version,
+            "pbkdf_args": pbkdf_args,
+            "escrow_cert": escrow_cert,
+            "add_backup_passphrase": request.backup_passphrase_enabled,
+            "min_luks_entropy": MIN_CREATE_ENTROPY,
```

Comment:
This data structure is required by Blivet, so I don't see the point of doing that unless Blivet provides us these constants. Anyway, this is a common way how to propagate arguments to Blivet's classes and functions and I don't really want to mess with it.

Reply:
I'm not saying changing the dict but using constants instead of plain strings. Pylint will find typo in the constant but no-one is able to do that for you in the plain strings.